### PR TITLE
Move all IInteractiveWindowOperations implementations to UIThreadOnly

### PR DESCRIPTION
**Pull request by amcasey**
They all intuitively correspond to UI operations and are not intended to be run in parallel with each other.

This is the first part of  #4044.

### CODE REVIEW

**Comment by amcasey** [comment]
FYI @tmat @cston @ManishJayaswal @KevinH-MS @DinoV

File: src/InteractiveWindow/Editor/InteractiveWindow.cs [hunk 1]
```diff
@@ -289,97 +282,27 @@ internal void AppendOutput(IEnumerable<string> output)
...
-                if (string.IsNullOrWhiteSpace(search))
-                {
-                    // don't store search as an uncommitted history item
-                    StoreUncommittedInputForHistory();
-                }
-
-                SetActiveCodeToHistory(previous);
-                UIThread(uiOnly => uiOnly.EditorOperations.MoveToEndOfDocument(false));
-            }
+            UIThread(uiOnly => uiOnly.HistoryPrevious(search));
```
Comment (cston) [inline src/InteractiveWindow/Editor/InteractiveWindow.cs]: HistoryPrevious is now asynchronous. In general, should these methods return Tasks so the caller can schedule continuations?

File: src/InteractiveWindow/Editor/InteractiveWindow.cs [hunk 2]
```diff
@@ -66,7 +62,6 @@ private enum ReplSpanKind
         private bool _adornmentToMinimize;
 
         private readonly IWpfTextView _textView;
```
Comment (cston) [inline src/InteractiveWindow/Editor/InteractiveWindow.cs]: Are there fields in this class that can be used on any thread? If not, perhaps uiOnly should be an IInteractiveWindow implementation containing all state, and the outer instance simply delegates to the inner instance, after switching to the UI thread.

File: src/InteractiveWindow/Editor/InteractiveWindow.cs [hunk 3]
```diff
@@ -66,7 +62,6 @@ private enum ReplSpanKind
...
-        private readonly History _history;
```
Comment (cston) [inline src/InteractiveWindow/Editor/InteractiveWindow.cs]: Are there any fields in InteractiveWindow that can be used safely on multiple threads? If not, perhaps uiOnly should be an IInteractiveWindow implementation containing all state, and the outer instance simply delegates to the inner instance, after switching to the UI thread.

Comment (amcasey) [inline src/InteractiveWindow/Editor/InteractiveWindow.cs]: I suspect that's where we'll end up, but I'm trying to break it down into logical pieces so the consequences are more obvious.

File: src/InteractiveWindow/Editor/InteractiveWindow.cs (hunk 1 above)
Comment (amcasey) [inline src/InteractiveWindow/Editor/InteractiveWindow.cs]: I'm not sure I understand.  How is ```HistoryPrevious``` more asynchronous than it was?

Comment (cston) [inline src/InteractiveWindow/Editor/InteractiveWindow.cs]: True, the selection change was asynchronous previously although the edits were synchronous. In either case, with async changes, unless we return a Task, the caller cannot avoid race conditions.

Comment (cston) [inline src/InteractiveWindow/Editor/InteractiveWindow.cs]: My mistake, UIThread() is synchronous as you mentioned offline.

**Comment by amcasey** [comment]
Test this please.

**Pull request merged**